src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,159 @@
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents Storage used to save/load data from Duke to disk.
+ */
+public class Storage {
+    /**
+     * Represents file path of storage.
+     */
+    private String filePath;
+
+    /**
+     * Constructor of Storage. Save/load file to input path.
+     * @param filePath File path to save/load file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads TaskList data from disk.
+     * @return TaskList loaded from disk.
+     * @throws LoadingErrorDukeException If no file found.
+     */
+    public List<Task> loadList() throws LoadingErrorDukeException {
+        List<Task> loadedList = new ArrayList<>();
+        try {
+            FileReader fileReader = new FileReader(filePath);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String readerLine = bufferedReader.readLine(); //initialise first line
+
+            while (readerLine != null) {
+                String[] readArray = readerLine.split("\\s\\|\\s");
+                Task task = createTaskFromInput(readArray);
+                loadedList.add(task);
+                readerLine = bufferedReader.readLine();
+            }
+            bufferedReader.close();
+        } catch (FileNotFoundException e) {
+            throw new LoadingErrorDukeException();
+        } catch (IOException | EmptyTaskDukeException | InvalidTaskDukeException e) {
+            e.printStackTrace();
+        }
+        return loadedList;
+    }
+
+    /**
+     * Saves TaskList data from Duke to disk.
+     * @param tasks Takes Tasks from TaskList and saves to disk.
+     */
+    public void saveList(TaskList tasks) {
+        File file = new File(filePath);
+        file.getParentFile().mkdirs();
+        try {
+            FileWriter fileWriter = new FileWriter(filePath);
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+            while (!tasks.isEmpty()) {
+                Task task = tasks.remove(1);
+                String line = generateLineFromTask(task);
+                bufferedWriter.write(line);
+                bufferedWriter.newLine();
+            }
+            bufferedWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Abstraction used to create tasks from String in data.
+     * @param inputArray String from reader.
+     * @return Task created from String.
+     * @throws InvalidTaskDukeException If input is invalid.
+     * @throws EmptyTaskDukeException If task is invalid.
+     */
+    private static Task createTaskFromInput(String[] inputArray) throws InvalidTaskDukeException, EmptyTaskDukeException {
+        Task createdTask = null;
+        switch(inputArray[0]) {
+        case "T":
+            createdTask = new ToDo(inputArray[2]);
```

Comment:
Perhaps comments could be added here to specify what inputArray[2] (and in subsequent lines, inputArray[3] too) refers to. Could improve code readability.